src/main/java/bio/terra/cli/command/resource/create/AwsSageMakerNotebook.java

```diff
@@ -0,0 +1,79 @@
+package bio.terra.cli.command.resource.create;
+
+import bio.terra.cli.command.shared.WsmBaseCommand;
+import bio.terra.cli.command.shared.options.ControlledResourceCreation;
+import bio.terra.cli.command.shared.options.Format;
+import bio.terra.cli.command.shared.options.WorkspaceOverride;
+import bio.terra.cli.serialization.userfacing.input.CreateAwsSageMakerNotebookParams;
+import bio.terra.cli.serialization.userfacing.input.CreateResourceParams;
+import bio.terra.cli.serialization.userfacing.resource.UFAwsSageMakerNotebook;
+import bio.terra.workspace.model.AccessScope;
+import bio.terra.workspace.model.StewardshipType;
+import picocli.CommandLine;
+
+/**
+ * This class corresponds to the fourth-level "terra resource create sagemaker-notebook" command.
+ */
+@CommandLine.Command(
+    name = "sagemaker-notebook",
+    description = "Add a controlled AWS SageMaker Notebook instance.",
+    showDefaultValues = true,
+    sortOptions = false)
+public class AwsSageMakerNotebook extends WsmBaseCommand {
+  @CommandLine.Mixin ControlledResourceCreation controlledResourceCreationOptions;
+  @CommandLine.Mixin WorkspaceOverride workspaceOption;
+  @CommandLine.Mixin Format formatOption;
+
+  @CommandLine.Option(
+      names = "--instance-name",
+      description =
+          "Name of the SageMaker Notebook instance. If not provided, a unique folder name will be generated (https://docs.aws.amazon.com/sagemaker/latest/APIReference/API_CreateNotebookInstance.html#sagemaker-CreateNotebookInstance-request-NotebookInstanceName).")
```

Comment:
1) unique notebook name
2) common with GCP notebook. Can this be moved to instanceName option

Reply:
done